test/node/bson_type_classes.test.ts

```diff
@@ -0,0 +1,100 @@
+import { expect } from 'chai';
+import { __isWeb__ } from '../register-bson';
+import {
+  Binary,
+  BSONRegExp,
+  BSONSymbol,
+  Code,
+  DBRef,
+  Decimal128,
+  Double,
+  Int32,
+  Long,
+  MaxKey,
+  MinKey,
+  ObjectId,
+  Timestamp,
+  UUID,
+  BSONValue
+} from '../register-bson';
+
+const BSONTypeClasses = [
+  Binary,
+  Code,
+  DBRef,
+  Decimal128,
+  Double,
+  Int32,
+  Long,
+  MinKey,
+  MaxKey,
+  ObjectId,
+  BSONRegExp,
+  BSONSymbol,
+  Timestamp,
+  UUID
+];
+
+const BSONTypeClassCtors = new Map<string, () => BSONValue>([
+  ['Binary', () => new Binary()],
```

Comment:
Is there a reason that these need to be factory functions and we can't just hard-code instances of the classes?

Reply:
If the constructors start throwing the tests crash on launch instead of inside the it block.